src/core/coap/coap.cpp

```diff
@@ -129,25 +132,92 @@ otError CoapBase::Send(ot::Message &aMessage, const Ip6::MessageInfo &aMessageIn
     return mSender(*this, aMessage, aMessageInfo);
 }
 
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+otError CoapBase::SendMessage(Message &                   aMessage,
+                              const Ip6::MessageInfo &    aMessageInfo,
+                              const TxParameters &        aTxParameters,
+                              ResponseHandler             aHandler,
+                              void *                      aContext,
+                              otCoapBlockwiseTransmitHook aTransmitHook,
+                              otCoapBlockwiseReceiveHook  aReceiveHook)
+#else  // OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
 otError CoapBase::SendMessage(Message &               aMessage,
                               const Ip6::MessageInfo &aMessageInfo,
                               const TxParameters &    aTxParameters,
                               ResponseHandler         aHandler,
                               void *                  aContext)
+#endif // OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
 {
     otError  error;
     Message *storedCopy = nullptr;
     uint16_t copyLength = 0;
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+    uint8_t  buf[kMaxBlockLength] = {0};
+    uint16_t bufLen               = kMaxBlockLength;
+    bool     moreBlocks           = false;
+#endif // OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
 
     switch (aMessage.GetType())
     {
     case OT_COAP_TYPE_ACKNOWLEDGMENT:
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+        // Check for block-wise transfer
+        if (aTransmitHook != nullptr)
+        {
+            // Check for Block2 option in header
+            if (aMessage.ReadBlockOptionValues(OT_COAP_OPTION_BLOCK2) == OT_ERROR_NONE)
+            {
+                // Check if it is the first block of the transfer
+                if (aMessage.GetBlockWiseBlockNumber() == 0)
```

Comment:
How about combining the nested three `if`s into one with conditions added using `&&`.
```suggestion
        if ((aTransmitHook != nullptr) && (aMessage.ReadBlockOptionValues(OT_COAP_OPTION_BLOCK2) == OT_ERROR_NONE) && (aMessage.GetBlockWiseBlockNumber() == 0))
        {

```